### PR TITLE
Add editable gas to all transactions

### LINKED ISSUE #85: Set gas/gas price of transactions
Add support to gas price in UI & Main Process

### PR DESCRIPTION
**Pull request by ianaya89**
Resolves #85 

*Is still pending add gas support to auction.*

### CODE REVIEW

File: public/main/settings/defaultSettings.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "app": {
     "node": {
-      "websocketApiUrl": "ws://parity.bloqrock.net:8546"
+      "websocketApiUrl": "ws://localhost:8546"
```
Comment (pablen) [inline public/main/settings/defaultSettings.json]: should we revert this change?

File: src/components/ConvertETHtoMTNForm.js [hunk 2]
```diff
@@ -48,10 +61,23 @@ class ConvertETHtoMTNForm extends React.Component {
...
+    showGasFields: false,
+    gasPrice: '1',
+    gasLimit: '21000',
     errors: {},
     error: null
   }
 
+  componentDidMount() {
+    sendToMainProcess('get-gas-price', {}).then(({ gasPrice }) => {
+      this.setState({ gasPrice: (gasPrice / 1000000000).toString() })
```
Comment (pablen) [inline src/components/ConvertETHtoMTNForm.js]: should we use BigNumber here?

File: src/components/ConvertETHtoMTNForm.js [hunk 3]
```diff
@@ -48,10 +61,23 @@ class ConvertETHtoMTNForm extends React.Component {
...
     status: 'init',
+    showGasFields: false,
+    gasPrice: '1',
+    gasLimit: '21000',
     errors: {},
     error: null
   }
 
+  componentDidMount() {
+    sendToMainProcess('get-gas-price', {}).then(({ gasPrice }) => {
```
Comment (pablen) [inline src/components/ConvertETHtoMTNForm.js]: this needs a `catch()` in order to avoid unhandled promise rejection error in renderer

File: src/components/ConvertETHtoMTNForm.js [hunk 4]
```diff
@@ -60,6 +86,21 @@ class ConvertETHtoMTNForm extends React.Component {
...
   }
 
+  onInputBlur = e => {
+    const { ethAmount } = this.state
+
+    if (!ethAmount || !isWeiable(ethAmount)) {
+      return
+    }
+
+    sendToMainProcess('metronome-convert-eth-gas-limit', {
```
Comment (pablen) [inline src/components/ConvertETHtoMTNForm.js:81]: also needs a `catch()`

File: src/components/ConvertMTNtoETHForm.js [hunk 5]
```diff
@@ -46,10 +58,23 @@ class ConvertMTNtoETHForm extends React.Component {
...
     password: null,
     status: 'init',
+    showGasFields: false,
+    gasPrice: '1',
+    gasLimit: '21000',
     errors: {},
     error: null
   }
 
+  componentDidMount() {
```
Comment (pablen) [inline src/components/ConvertMTNtoETHForm.js:54]: Same as above ( catch and BigNumber doubt)

File: src/components/ConvertMTNtoETHForm.js [hunk 6]
```diff
@@ -64,21 +89,38 @@ class ConvertMTNtoETHForm extends React.Component {
...
+    const { mtnAmount } = this.state
+
+    if (!mtnAmount || !isWeiable(mtnAmount)) {
+      return
+    }
+
+    sendToMainProcess('metronome-convert-met-gas-limit', {
+      from: this.props.from,
+      value: Web3.utils.toWei(mtnAmount.replace(',', '.'))
+    }).then(({ gasLimit }) => {
```
Comment (pablen) [inline src/components/ConvertMTNtoETHForm.js]: Same as above (catch)

File: src/components/ConvertETHtoMTNForm.js [hunk 7]
```diff
@@ -163,6 +212,50 @@ class ConvertETHtoMTNForm extends React.Component {
                 id="password"
               />
             </Sp>
+
+            <Sp mt={4} mb={2}>
+              <Flex.Row justify="space-between">
+                <Flex.Item grow="1" basis="0">
+                  {showGasFields ? (
+                    <TextInput
+                      type="number"
```
Comment (pablen) [inline src/components/ConvertETHtoMTNForm.js:207]: Minor nitpick: as we're using a `number` type field for `gasLimit` and it must be a positive integer, we could add a `min="0"` attribute to avoid negative values and a `step="1"` to avoid decimal values.

File: src/components/ConvertETHtoMTNForm.js [hunk 8]
```diff
@@ -163,6 +212,50 @@ class ConvertETHtoMTNForm extends React.Component {
...
+                    <GasLabel>Gas Limit: {gasLimit} (UNITS)</GasLabel>
+                  )}
+                </Flex.Item>
+
+                {showGasFields && <Sp mt={6} mx={1} />}
+
+                <Flex.Item grow="1" basis="0">
+                  {showGasFields ? (
+                    <TextInput
+                      type="number"
```
Comment (pablen) [inline src/components/ConvertETHtoMTNForm.js:225]: Same as above here `min="0"` (but no `step`)

File: public/main/settings/defaultSettings.json (hunk 1 above)
Comment (gabmontes) [reply public/main/settings/defaultSettings.json]: @ianaya89 we should...

File: src/components/ConvertETHtoMTNForm.js (hunk 2 above)
Comment (gabmontes) [reply src/components/ConvertETHtoMTNForm.js]: Gas price would not be that high to require BN but yeah, we should use BN for any math regarding values.

**Review by pablen — APPROVED** [review]

**Review by gabmontes — APPROVED** [review]
